src/sentry/workflow_engine/endpoints/project_detector_details.py

```diff
@@ -69,3 +72,24 @@ def get(self, request: Request, project, detector):
             DetectorSerializer(),
         )
         return Response(serialized_detector)
+
+    @extend_schema(
+        operation_id="Delete a Detector",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            GlobalParams.PROJECT_ID_OR_SLUG,
+            DetectorParams.DETECTOR_ID,
+        ],
+        responses={
+            204: RESPONSE_NO_CONTENT,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def delete(self, request: Request, project: Project, detector: Detector):
```

Comment:
could we prevent the deletion of detectors with `type=ErrorGroupType.slug`? every project should have an error detector, and these should only be deleted upon project deletion

Reply:
I forget whether we can delete other "one of a kind" detectors like n+1 etc.

I think that this should be defined on the `GroupType`. `GroupType.required_for_project` or something? Ideally, `workflow_engine` should avoid importing specific detectors as much as possible.